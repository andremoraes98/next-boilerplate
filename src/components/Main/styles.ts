'use client'

import styled from 'styled-components'

export const Wrapper = styled.main`
  background-color: #f0f0f0;
  height: 100vh;
  display: flex;
  justify-content: center;
  align-items: center;
`
