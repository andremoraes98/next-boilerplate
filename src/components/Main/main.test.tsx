import { render, screen } from '@testing-library/react'

import Main from '.'

describe('Main', () => {
  it('should render the heading', () => {
    const { container } = render(<Main />)

    const heading = screen.getByRole('heading', { name: /hello world/i })

    expect(heading).toBeInTheDocument()
    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render the background color correctly', () => {
    const { container } = render(<Main />)

    expect(container.firstChild).toHaveStyle({ 'background-color': '#f0f0f0' })
  })
})
