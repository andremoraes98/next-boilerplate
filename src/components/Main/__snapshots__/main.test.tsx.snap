// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main should render the heading 1`] = `
.c0 {
  background-color: #f0f0f0;
  height: 100vh;
  display: flex;
  justify-content: center;
  align-items: center;
}

<main
  class="c0"
>
  <h1>
    Hello World
  </h1>
</main>
`;
